felix/bpf/proxy/proxy.go

```diff
@@ -337,3 +337,20 @@ type loggerRecorder struct{}
 
 func (r *loggerRecorder) Eventf(regarding runtime.Object, related runtime.Object, eventtype, reason, action, note string, args ...interface{}) {
 }
+
+func InvokeCategorizeEndpoints(svcMap k8sp.ServiceMap, epsMap k8sp.EndpointsMap, nodeLabels map[string]string) (clusterEndpoints, localEndpoints, allReachableEndpoints []k8sp.Endpoint, hasAnyEndpoints bool) {
+
+	// TODO - correct this because if the map has more than one entry then only the last entry would be used!
+	var svcInfo k8sp.ServicePort
+	for k := range svcMap {
+		svcInfo = svcMap[k]
+	}
+
+	var endpoints []k8sp.Endpoint
+	for k := range epsMap {
+		endpoints = epsMap[k]
+	}
```

Comment:
We do need to do this rewrite ourselves? Does not k8s provide it? It might be that k8s proxy switched to these structures and it would be better for us to change our code as well :man_shrugging:

Reply:
At a higher level the K8s code does work with the svcMap and epsMap objects but the [proxier.go](https://github.com/kubernetes/kubernetes/blob/b1d98bd7b34da70bd37187aa88b662e0d047d0ff/pkg/proxy/iptables/proxier.go#L997) code iterates thru the map key/val pairs invokes the `CategorizeEndpoints` passing in each svc and eps value for the corresponding key but I was trying to prototype invoking `CategorizeEndpoints` once in isolation to be unit tested